From leitao@linux.vnet.ibm.com  Thu Oct  8 15:40:52 2009
From: Breno Leitão <leitao@linux.vnet.ibm.com>
Date: Thu, 24 Sep 2009 16:58:22 -0300
Subject: jsm: IRQ handlers doesn't need to have IRQ_DISABLED enabled
To: greg@kroah.com
Cc: linux-kernel@vger.kernel.org, scottk@digi.com
Message-ID: <39629f59067510a052f4a493be9044a572de7f15.1253820690.git.leitao@linux.vnet.ibm.com>


Currently jsm is showing the following message when loaded:

IRQ 432/JSM: IRQF_DISABLED is not guaranteed on shared IRQs

It's because the request_irq() is called using IRQF_DISABLED
and IRQF_SHARED.
Actually there is no need to use IRQF_DISABLED in this driver.

Signed-off-by: Breno Leitão <leitao@linux.vnet.ibm.com>
Cc: Scott Kilau <scottk@digi.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/jsm/jsm_driver.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/jsm/jsm_driver.c
+++ b/drivers/serial/jsm/jsm_driver.c
@@ -123,7 +123,7 @@ static int __devinit jsm_probe_one(struc
 	}
 
 	rc = request_irq(brd->irq, brd->bd_ops->intr,
-			IRQF_DISABLED|IRQF_SHARED, "JSM", brd);
+			IRQF_SHARED, "JSM", brd);
 	if (rc) {
 		printk(KERN_WARNING "Failed to hook IRQ %d\n",brd->irq);
 		goto out_iounmap;
